File: app/product.ts
export class Product {
    public productId:string;
    public productName:string;
    public productPrice: number;
    public productUrl:string;
    public categoryType:number;
    public productDescription:string;
    public productStock:number;
    constructor(
        productId:string='',
        productName:string='',
        productPrice: number=-1,
        productUrl:string='',
        categoryType:number=-1,
        productDescription:string='',
        productStock:number=-1
    ){
        this.productId=productId;
        this.productName=productName;
        this.productPrice=productPrice;
        this.productStock=productStock;
        this.productUrl=productUrl;
        this.categoryType=categoryType;
        this.productDescription=productDescription;
    }
}

File: src/main/java/com/wipro/capstoneshopfrohome/entity/Product.java
package com.wipro.capstoneshopfrohome.entity;

import java.math.BigDecimal;

import javax.persistence.Entity;
import javax.persistence.Id;

import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.DynamicUpdate;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@DynamicUpdate
public class Product {

	@Id
	private String productId;
	private String productName;
	private BigDecimal productPrice;
	private Integer productStock;
	private String productDescription;
	private String productUrl;
	
	@ColumnDefault("0")
	private Integer productStatus;
	
	@ColumnDefault("0")
	private Integer categoryType;

}